packages/react/src/components/TimePicker/TimePicker.tsx

```diff
@@ -42,43 +47,111 @@ export const TimePicker: React.FunctionComponent<ITimePickerProps> = ({
   timeRange,
   strings = getDefaultStrings(useHour12, showSeconds),
   defaultValue,
+  value,
+  dateAnchor,
   onChange,
   onFormatDate,
   onValidateUserInput,
+  onGetErrorMessage,
+  placeholder = strings.defaultTimePickerPlaceholder,
   ...rest
 }: ITimePickerProps) => {
-  const [userText, setUserText] = React.useState<string>('');
+  const [comboBoxText, setComboBoxText] = React.useState<string>('');
+  const [selectedKey, setSelectedKey] = React.useState<string | number | undefined>();
   const [errorMessage, setErrorMessage] = React.useState<string>('');
 
+  const fallbackDateAnchor = useConst(new Date());
+
+  const [dateStartAnchor, setDateStartAnchor] = React.useState<Date>(dateAnchor || defaultValue || new Date());
+  const [dateEndAnchor, setDateEndAnchor] = React.useState<Date>(dateAnchor || defaultValue || new Date());
+
+  const [selectedTime, setSelectedTime] = useControllableValue(value, defaultValue, (_ev: any, newTime: Date) =>
+    onChange?.(undefined, newTime),
+  );
```

Comment:
Sorry to catch this late, could you change this so that the onChange doesn't need to have optional arguments?
```suggestion
  const [selectedTime, setSelectedTime] = useControllableValue(value, defaultValue, (ev: React.FormEvent<IComboBox>, newTime: Date) =>
    onChange?.(ev, newTime),
  );
```

This came up after talking to @behowell offline.

Reply:
Reverted the onChange parameters back to their original typings.